File: leetcode_75/binary_tree_DFS/maximum_depth_of_binary_tree_104.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        left_depth = self.maxDepth(root.left)
        right_depth = self.maxDepth(root.right)
        return max(left_depth, right_depth) + 1

        # temp = []
        # while root:
        #     # temp
        #     if root.left:
        #         maxDepth(root.left)
        #         temp +=1
        #     if root.right:
        #         maxDepth(root.right)
        #         temp +=1
        #     if not root.left and maxDepth(root.right):
        #         temp = max(temp)

File: leetcode_75/binary_tree_DFS/maximum_depth_of_binary_tree_104.java
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        // if (root == null) return 0;
        // 可以寫成一行的

        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        return Math.max(left, right) + 1;
    }
}